File: BCC221/semana06-tp01/Chacara.cpp
#include "Chacara.h"

Chacara :: Chacara(const int id, const float valor, const string proprietario,
                        const string rua, const string bairro, const string cidade, const int numero,
                        const int quartos, const int banheiros, const bool salao_festa,const bool salao_jogos,
                        const bool campo_futebol,const bool churrasqueira,const bool piscina): Imovel(id, valor, proprietario, rua, bairro,
                        cidade, numero, quartos, banheiros), salao_festa(salao_festa), salao_jogos(salao_jogos),
                        campo_futebol(campo_futebol), churrasqueira(churrasqueira), piscina(piscina){}

bool Chacara::getSalaoFesta() const{
    return salao_festa;
}

void Chacara::setSalaoFesta(bool salao_festa){
    this->salao_festa = salao_festa;
}

bool Chacara::getSalaoJogos() const{
    return salao_jogos;
}

void Chacara::setSalaoJogos(bool salao_jogos){
    this->salao_jogos = salao_jogos;
}

bool Chacara::getCampoFutebol() const{
    return campo_futebol;
}

void Chacara::setCampoFutebol(bool campo_futebol){
    this->campo_futebol = campo_futebol;
}

bool Chacara::getChurrasqueira() const{
    return churrasqueira;
}

void Chacara::setChurrasqueira(bool churrasqueira){
    this->churrasqueira = churrasqueira;
}

bool Chacara::getPiscina() const{
    return piscina;
}

void Chacara::setPiscina(bool piscina){
    this->piscina = piscina;
}

void Chacara::saida(ostream& out){
    Imovel::saida(out);
    if (this->getPiscina()){
        out << "    Piscina: SIM" << endl;
    }
    else{
        out << "    Piscina: NAO" << endl;
    }
}

ostream& operator<<(ostream& out, Chacara& obj){
    obj.saida(out);
	return out;
}

void Chacara::imprimirTudo(){
    cout << "    Salao de festas: " << (this->getSalaoFesta() ? "SIM" : "NAO");
    cout << "\n    Salao de jogos: " << (this->getSalaoJogos() ? "SIM" : "NAO");
    cout << "\n    Campo de futebol: " << (this->getCampoFutebol() ? "SIM" : "NAO");
    cout << "\n    Churrasqueira: " << (this->getChurrasqueira() ? "SIM" : "NAO") << endl;
}

Chacara::~Chacara() = default;

File: BCC221/semana14-tp02/src/main/java/backend/Chacara.java
package backend;

public class Chacara extends Imovel {

    private boolean salao_festas;
    private boolean salao_jogos;
    private boolean campo_futebol;
    private boolean churrasqueira;
    private boolean piscina;

    //Construtor geral
    public Chacara(double valor, String proprietario, String rua, String bairro, String cidade, int numero, int quartos, int banheiros,
            boolean salao_festas, boolean salao_jogos, boolean campo_futebol, boolean churrasqueira, boolean piscina) {
        super(valor, proprietario, rua, bairro, cidade, numero, quartos, banheiros);
        this.salao_festas = salao_festas;
        this.salao_jogos = salao_jogos;
        this.campo_futebol = campo_futebol;
        this.churrasqueira = churrasqueira;
        this.piscina = piscina;
    }

    //Construtor utilizando imovel e demais atributos
    public Chacara(Imovel imovel, boolean salao_festas, boolean salao_jogos, boolean campo_futebol, boolean churrasqueira, boolean piscina) {
        super(imovel);
        this.salao_festas = salao_festas;
        this.salao_jogos = salao_jogos;
        this.campo_futebol = campo_futebol;
        this.churrasqueira = churrasqueira;
        this.piscina = piscina;
    }

    //Construtor inicializando padrões
    public Chacara() {
        super(0.0, "", "", "", "", 0, 0, 0);
        this.salao_festas = false;
        this.salao_jogos = false;
        this.campo_futebol = false;
        this.churrasqueira = false;
        this.piscina = false;
    }

    public boolean isSalao_festas() {
        return salao_festas;
    }

    public void setSalao_festas(boolean salao_festas) {
        this.salao_festas = salao_festas;
    }

    public boolean isSalao_jogos() {
        return salao_jogos;
    }

    public void setSalao_jogos(boolean salao_jogos) {
        this.salao_jogos = salao_jogos;
    }

    public boolean isCampo_futebol() {
        return campo_futebol;
    }

    public void setCampo_futebol(boolean campo_futebol) {
        this.campo_futebol = campo_futebol;
    }

    public boolean isChurrasqueira() {
        return churrasqueira;
    }

    public void setChurrasqueira(boolean churrasqueira) {
        this.churrasqueira = churrasqueira;
    }

    public boolean isPiscina() {
        return piscina;
    }

    public void setPiscina(boolean piscina) {
        this.piscina = piscina;
    }

    //Sobreescrita do método toString
    @Override
    public String toString() {
        return super.toString() + "Piscina: " + (this.piscina ? "SIM" : "NAO");
    }

    //Sobreescrita do método que imprime atributos específicos da classe
    @Override
    public String imprimirClasse() {
        return super.toString() + "\n\tCaracterística Especifica\n"
                + "\t\tSalão de festas: " + (this.salao_festas ? "SIM" : "NAO")
                + "\n\t\tSalão de jogos: " + (this.salao_jogos ? "SIM" : "NAO")
                + "\n\t\tCampo de futebol: " + (this.campo_futebol ? "SIM" : "NAO")
                + "\n\t\tChurrasqueira: " + (this.churrasqueira ? "SIM" : "NAO");
    }
}
